[CI] EsqlSpecIT test {lookup-join.LookupIPFromIndexKeep ASYNC} failing

**Build Scans:**
- [elasticsearch-intake #13857 / part3](https://gradle-enterprise.elastic.co/s/qoeupwqtl3k72)
- [elasticsearch-intake #13811 / part3](https://gradle-enterprise.elastic.co/s/wp3drwdirkexm)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT" -Dtests.method="test {lookup-join.LookupIPFromIndexKeep ASYNC}" -Dtests.seed=BE0A9C1F9174B2FD -Dtests.locale=fr-MQ -Dtests.timezone=Asia/Riyadh -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Blookup-join.LookupIPFromIndexKeep%20ASYNC%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Data mismatch:
row 0 column 0:a list containing
row 0 column 0:0: expected "2023-10-23T13:55:01.543Z" but was "2023-10-23T12:27:28.948Z"
row 0 column 1:a list containing
row 0 column 1:0: expected "172.21.3.15" but was "172.21.2.113"
row 0 column 2:a list containing
row 0 column 2:0: expected "1756467" but was "2764889"
row 0 column 3:a list containing
row 0 column 3:0: expected "Connected to 10.1.0.1" but was "Connected to 10.1.0.2"
row 0 column 4:a list containing
row 0 column 4:0: expected "Production" but was "QA"
row 1 column 0:a list containing
row 1 column 0:0: expected "2023-10-23T13:53:55.832Z" but was "2023-10-23T12:15:03.360Z"
row 1 column 1:a list containing
row 1 column 1:0: expected "172.21.3.15" but was "172.21.2.162"
row 1 column 2:a list containing
row 1 column 2:0: expected "5033755" but was "3450233"
row 1 column 3:a list containing
row 1 column 3:0: expected "Connection error" but was "Connected to 10.1.0.3"
row 1 column 4:a list containing
row 1 column 4:0: expected "Production" but was "QA"
row 2 column 0:a list containing
row 2 column 0:0: expected "2023-10-23T13:52:55.015Z" but was "2023-10-23T13:55:01.543Z"
```

**Issue Reasons:**
- [main] 2 failures in test test {lookup-join.LookupIPFromIndexKeep ASYNC} (1.3% fail rate in 149 executions)
- [main] 2 failures in step part3 (6.9% fail rate in 29 executions)
- [main] 2 failures in pipeline elasticsearch-intake (6.9% fail rate in 29 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -239,20 +239,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.RegressionIT
   method: testTwoJobsWithSameRandomizeSeedUseSameTrainingSet
   issue: https://github.com/elastic/elasticsearch/issues/117805
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {lookup-join.LookupIPFromIndexKeep ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/117974
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {lookup-join.LookupMessageFromIndex ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/117975
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {lookup-join.LookupMessageFromIndexKeep ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/117976
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {lookup-join.LookupMessageFromIndex SYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/117980
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  issue: https://github.com/elastic/elasticsearch/issues/117981
 - class: org.elasticsearch.packaging.test.ArchiveGenerateInitialCredentialsTests
   method: test30NoAutogenerationWhenDaemonized
   issue: https://github.com/elastic/elasticsearch/issues/117956

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join.csv-spec
@@ -127,6 +127,7 @@ FROM sample_data
 | EVAL client_ip = client_ip::keyword
 | LOOKUP JOIN clientips_lookup ON client_ip
 ;
+ignoreOrder:true;
 
 @timestamp:date          | event_duration:long | message:keyword       | client_ip:keyword | env:keyword
 2023-10-23T13:55:01.543Z | 1756467             | Connected to 10.1.0.1 | 172.21.3.15       | Production
@@ -146,6 +147,7 @@ FROM sample_data
 | LOOKUP JOIN clientips_lookup ON client_ip
 | KEEP @timestamp, client_ip, event_duration, message, env
 ;
+ignoreOrder:true;
 
 @timestamp:date          | client_ip:keyword | event_duration:long | message:keyword       | env:keyword
 2023-10-23T13:55:01.543Z | 172.21.3.15       | 1756467             | Connected to 10.1.0.1 | Production
@@ -230,6 +232,7 @@ required_capability: join_lookup_v4
 FROM sample_data
 | LOOKUP JOIN message_types_lookup ON message
 ;
+ignoreOrder:true;
 
 @timestamp:date          | client_ip:ip | event_duration:long | message:keyword       | type:keyword
 2023-10-23T13:55:01.543Z | 172.21.3.15  | 1756467             | Connected to 10.1.0.1 | Success
@@ -248,6 +251,7 @@ FROM sample_data
 | LOOKUP JOIN message_types_lookup ON message
 | KEEP @timestamp, client_ip, event_duration, message, type
 ;
+ignoreOrder:true;
 
 @timestamp:date          | client_ip:ip | event_duration:long | message:keyword       | type:keyword
 2023-10-23T13:55:01.543Z | 172.21.3.15  | 1756467             | Connected to 10.1.0.1 | Success

